braindecode/datautil/hub_formats.py

```diff
@@ -0,0 +1,464 @@
+        Specific recording IDs to load. If None, loads all.
+
+    Returns
+    -------
+    BaseConcatDataset
+        The loaded dataset.
+
+    Examples
+    --------
+    >>> # Load from local zarr directory
+    >>> dataset = load_from_zarr("dataset.zarr", preload=True)
+    >>>
+    >>> # Load from Hugging Face Hub (handled automatically)
+    >>> from braindecode.datasets import BaseConcatDataset
+    >>> dataset = BaseConcatDataset.from_pretrained("username/dataset-name")
+    """
+    if not ZARR_AVAILABLE:
+        raise ImportError(
+            "Zarr is not installed. Install with: pip install zarr"
+        )
+
+    # Import at runtime to avoid circular dependency
+    from ..datasets.base import BaseConcatDataset
+
+    input_path = Path(input_path)
+
+    if not input_path.exists():
+        raise FileNotFoundError(f"{input_path} does not exist.")
+
+    # Open zarr store
+    store = zarr.DirectoryStore(input_path)
+    root = zarr.group(store=store)
+
+    n_datasets = root.attrs["n_datasets"]
+    is_windowed = root.attrs["is_windowed"]
+
+    # Determine which datasets to load
+    if ids_to_load is None:
+        ids_to_load = list(range(n_datasets))
+
+    datasets = []
+    for i_ds in ids_to_load:
+        grp = root[f"recording_{i_ds}"]
+
+        if is_windowed:
+            ds = _load_windowed_dataset_zarr(grp, preload)
+        else:
+            ds = _load_raw_dataset_zarr(grp, preload)
+
+        datasets.append(ds)
+
+    # Create concat dataset
+    concat_ds = BaseConcatDataset(datasets)
+
+    # Restore preprocessing kwargs
+    for kwarg_name in [
+        "raw_preproc_kwargs",
+        "window_kwargs",
+        "window_preproc_kwargs",
+    ]:
+        if kwarg_name in root.attrs:
+            kwargs = json.loads(root.attrs[kwarg_name])
+            setattr(concat_ds, kwarg_name, kwargs)
+
+    return concat_ds
+
+
+def _load_windowed_dataset_zarr(grp: zarr.Group, preload: bool):
+    """Load a windowed dataset from Zarr group."""
+    import mne
+    # Import at runtime to avoid circular dependency
+    from ..datasets.base import WindowsDataset
+
+    # Load metadata
+    metadata = pd.read_json(grp.attrs["metadata"], orient="split")
+
+    # Load description
+    description = pd.read_json(grp.attrs["description"], typ="series")
+
+    # Load info
+    info_dict = json.loads(grp.attrs["info"])
+    info = _dict_to_mne_info(info_dict)
+
+    # Load data
+    if preload:
+        data = grp["data"][:]
+    else:
+        data = grp["data"][:]
+        warnings.warn(
+            "Lazy loading from Zarr not fully implemented yet. "
+            "Loading all data into memory.",
+            UserWarning,
+        )
+
+    # Create Epochs object
+    events = np.column_stack([
+        metadata["i_start_in_trial"].values,
+        np.zeros(len(metadata), dtype=int),
+        metadata["target"].values,
+    ])
+
+    epochs = mne.EpochsArray(data, info, events=events, metadata=metadata)
```

Comment:
This isn't right, everything is raw, but we make the the distinction with another way.